Mathlib/CategoryTheory/Generator/Basic.lean

```diff
@@ -177,6 +173,20 @@ theorem IsSeparating.isDetecting [Balanced C] {𝒢 : Set C} (h𝒢 : IsSeparati
     obtain ⟨t, rfl, -⟩ := hf G hG i
     rw [Category.assoc, hgh, Category.assoc]
 
+lemma IsDetecting.isIso_iff_of_mono {𝒢 : Set C} (h𝒢 : IsDetecting 𝒢)
+    {X Y : C} (f : X ⟶ Y) [Mono f] :
+    IsIso f ↔ ∀ (s : 𝒢), Function.Surjective ((coyoneda.obj (op s.1)).map f) := by
+  constructor
+  · intro h
+    rw [isIso_iff_yoneda_map_bijective] at h
+    rintro ⟨A, _⟩
+    exact (h A).2
+  · intro hf
+    refine h𝒢 _ (fun A hA g ↦ existsUnique_of_exists_of_unique ?_ ?_)
+    · exact hf ⟨A, hA⟩ g
+    · intro l₁ l₂ h₁ h₂
+      rw [← cancel_mono f, h₁, h₂]
+
```

Comment:
Please deduce the dual result about codetectors.

Reply:
Thanks for the suggestion.